Search input people

tomecky1 opened the pull request:
This pull request consists of my Input Search for people branch. I still do not know how to implement poeople searching feature into the SearchResults component. If you have any idea it would be nice.

tomecky1 committed:
Add people search feature with dynamic placeholder support

Introduce a new PeopleSearch component for searching people, including styled components and a custom hook for API requests. Update Navigation to support dynamic placeholders based on the current route.

tomecky1 committed:
Merge remote-tracking branch 'origin/searchInputPeople' into searchInputPeople

tomecky1 committed:
Add people search functionality to navigation and results

Extended the search feature to include people alongside movies. Updated navigation to handle dynamic placeholders, people search results, and person detail navigation. Adjusted search results page to render people or movies based on the current route.

tomecky1 committed:
Optimize debounce timing in search hooks

Reduced the debounce timeout from 500ms to 300ms in `useMovieSearch` and `usePeopleSearch` to improve responsiveness.

albert383 commented on src/common/Navigation/Navigation.js:
```diff
@@ -56,12 +75,32 @@ export const Navigation = () => {
           />
         </NavigationInput>
       </NavigationList>
+      {/* Wyświetlenie wyników dla osób */}
```
we can remove all comments before merge

albert383 commented on src/common/Navigation/Navigation.js:
```diff
@@ -56,12 +75,32 @@ export const Navigation = () => {
+        </ul>
+      )}
+
+      {/* Wyświetlenie wyników dla filmów */}
```
same here, we dont need any unnecessary comments

albert383 commented on src/features/PeopleSearch/index.js:
```diff
@@ -0,0 +1,37 @@
+  const [searchQuery, setSearchQuery] = useState("");
+  const navigate = useNavigate();
+  const { searchResults } = usePeopleSearch(searchQuery);
+
```
blankspace not needed

albert383 commented on src/features/SearchResults/index.js:
```diff
@@ -55,14 +56,27 @@ export const SearchResults = () => {
+  // const { searchResults } = useMovieSearch(query);
+  const location = useLocation()
+
+  // Wykrywanie aktualnej strony
```
same here, no comments required

albert383 commented on src/features/SearchResults/index.js:
```diff
@@ -55,14 +56,27 @@ export const SearchResults = () => {
 
   const [searchParams] = useSearchParams();
   const query = searchParams.get("query");
+  // const { searchResults } = useMovieSearch(query);
```
no comments required

albert383 commented on src/features/SearchResults/index.js:
```diff
@@ -55,14 +56,27 @@ export const SearchResults = () => {
+  const isMoviesPage = location.pathname.startsWith("/movies");
+  const isPeoplePage = location.pathname.startsWith("/people");
+
+  // Wyniki wyszukiwania: filmy lub osoby
```
no comments required

albert383 reviewed (changes requested):
Everythink fine excluding extra comments should be removed from final version

Krystian22FrontEnd commented on src/common/Navigation/Navigation.js:
```diff
@@ -35,6 +43,17 @@ export const Navigation = () => {
     }
   };
 
+  // Dynamiczny placeholder w zależności od aktywnej strony
+  const currentPlaceholder =
+    location.pathname === "/person"
+      ? "Search for people..."
+      : "Search for movies...";
```
Great idea. Less code, more readible.

tomecky1 committed:
Improve search functionality and clean up code comments

tomecky1 committed:
Reset search query on route change and update error logging.

Added a useEffect hook to clear the search input when the route changes, improving user experience during navigation. Replaced non-English text in error logging with an English equivalent for consistency.

albert383 reviewed (approved):
I think now everything is properly corrected!

tomecky1 committed:
Merge branch 'master' into searchInputPeople
